client/src/main/java/org/opensearch/ml/client/MachineLearningClient.java

```diff
@@ -480,8 +480,7 @@ default ActionFuture<DeleteResponse> deleteAgent(String agentId) {
      * @param listener a listener to be notified of the result
      */
     default void deleteAgent(String agentId, ActionListener<DeleteResponse> listener) {
-        PlainActionFuture<DeleteResponse> actionFuture = PlainActionFuture.newFuture();
-        deleteAgent(agentId, null, actionFuture);
+        deleteAgent(agentId, null, listener);
```

Comment:
will it trigger BWC issue by using a listener?

Reply:
nope. Please check other apis.